### PR TITLE
shan/289_regex_otp

### LINKED ISSUE #289: Apply Regex pattern to extract OTP
[Utils#generate_token_OTP](https://github.com/bombardier-gif/covid-vaccine-booking/blob/main/src/utils.py#L1023) can be improved to extract the OTP using a 6 digit regex pattern to accomodate changes to CoWin SMS in future. Maybe something like this:
```
import re

regex = r"(?<!\d)\d{6}(?!\d)"

test_str = "your otp is 495734. Good luck!"

matches = re.finditer(regex, test_str, re.MULTILINE)

for matchNum, match in enumerate(matches, start=1):
    
    print ("Match {matchNum} was found at {start}-{end}: {match}".format(matchNum = matchNum, start = match.start(), end = match.end(), match = match.group()))
    
    for groupNum in range(0, len(match.groups())):
        groupNum = groupNum + 1
        
        print ("Group {groupNum} found at {start}-{end}: {group}".format(groupNum = groupNum, start = match.start(groupNum), end = match.end(groupNum), group = match.group(groupNum)))
```

### PR DESCRIPTION
**Pull request by shan-96**
closes #289 

- added tests structure (can help in #301)
- added method to extract OTP from regex

### CODE REVIEW

**Comment by bombardier-gif** [comment]
Thanks @shan-96 I will merge this once our repo stablizes a bit

**Comment by 6mxo89sqh** [comment]
#351 and this are duplicate.

**Comment by 6mxo89sqh** [comment]
@shan-96 Please resolve the conflicts.

**Comment by shantanu-deriv** [comment]
> @shan-96 Please resolve the conflicts.

PR will be up for re-submission by tonight

**Comment by shan-96** [comment]
@6mxo89sqh conflicts are now resolved. Thanks

File: src/utils.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 from inputimeout import inputimeout, TimeoutOccurred
 import tabulate, copy, time, datetime, requests, sys, os, random
 from captcha import captcha_builder_manual, captcha_builder_auto
+import uuid
```
Comment (6mxo89sqh) [inline src/utils.py:7]: nit: Unused import

File: src/utils.py [hunk 2]
```diff
@@ -1118,9 +1120,8 @@ def generate_token_OTP(mobile, request_header, kvdb_bucket):
             print("OTP SMS is:" + response.text)
             print("OTP SMS len is:" + str(len(response.text)))
 
-            OTP = response.text
-            OTP = OTP.replace("Your OTP to register/access CoWIN is ", "")
-            OTP = OTP.replace(". It will be valid for 3 minutes. - CoWIN", "")
+            sms_regex = r"(?<!\d)\d{6}(?!\d)"
```
Comment (6mxo89sqh) [inline src/utils.py:1123]: Can you please move this variable at the top near other constants?
```suggestion
            SMS_REGEX = r"(?<!\d)\d{6}(?!\d)"
```

File: src/utils.py [hunk 3]
```diff
@@ -1118,9 +1120,8 @@ def generate_token_OTP(mobile, request_header, kvdb_bucket):
...
+            OTP = extract_from_regex(response.text, sms_regex)
```
Comment (6mxo89sqh) [inline src/utils.py:1124]:
```suggestion
            OTP = extract_from_regex(response.text, SMS_REGEX)
```

**Comment by 6mxo89sqh** [comment]
Tested; works fine. left a few comments to be fixed before merging.

**Review by 6mxo89sqh — APPROVED** [review]

**Pull request merged**